aw_core/transforms.py

```diff
@@ -132,70 +135,117 @@ def eventcount(self) -> int:
-                        chunks[event.data[chunk_key]]["data"][k]["duration"] += event.duration
-                    if v not in chunks[event.data[chunk_key]]["data"][k]["values"]:
-                        chunks[event.data[chunk_key]]["data"][k]["values"][v] = {}
-                        chunks[event.data[chunk_key]]["data"][k]["values"][v]["duration"] = copy(event.duration)
-                    else:
-                        chunks[event.data[chunk_key]]["data"][k]["values"][v]["duration"] += event.duration
-    # Package response
-    return ChunkResult(chunks, totduration, eventcount)
+                    chunk["data"][k]["duration"] += event.duration
+                    chunk["data"][k]["values"][v]["duration"] += event.duration
 
+    # Convert all the defaultdicts into normal dicts
+    chunks = dict(chunks)
+    if include_subchunks:
+        for chunk in chunks.values():
+            chunk["data"] = dict(chunk["data"])
+            for subchunk in chunk["data"].values():
+                subchunk["values"] = dict(subchunk["values"])
 
-def merge_chunks(chunk1, chunk2):
-    result = {}
-    for label in set(chunk1.keys()).union(set(chunk2.keys())):
-        if label in chunk1 and label in chunk2:
-            result[label] = {"data":{}}
-            c1kv = chunk1[label]["data"]
-            c2kv = chunk2[label]["data"]
-            for key in set(c1kv.keys()).union(set(c2kv.keys())):
-                if key in c1kv and key in c2kv:
-                    result[label]["data"][key] = {"values":{}}
-                    c1k = c1kv[key]["values"]
-                    c2k = c2kv[key]["values"]
-                    for val in set(c1k.keys()).union(set(c2k.keys())):
-                        if val in c1k and val in c2k:
-                            c1v = c1k[val]
-                            c2v = c2k[val]
-                            result[label]["data"][key]["values"][val] = {
-                                "duration":
-                                    c1v["duration"] +
-                                    c2v["duration"]
-                            }
-                        elif val in c1v:
-                            result[label]["data"][key]["values"][val] = c1k[val]
-                        elif val in c2v:
-                            result[label]["data"][key]["values"][val] = c2k[val]
-                elif key in c1kv:
-                    result[label]["data"][key] = c1kv[key]
-                elif key in c2kv:
-                    result[label]["data"][key] = c2kv[key]
-        elif label in chunk1:
-            result[label] = chunk1[label]
-        elif label in chunk2:
-            result[label] = chunk2[label]
+    # Package response
+    total_duration = reduce(operator.add, (event.duration for event in events), timedelta(0))
+    return ChunkResult(chunks, total_duration, len(events))
+
+
+def merge_chunks(chunk1: ChunkResult, chunk2: ChunkResult):
+    """What exactly is chunk1 and chunk2?"""
```

Comment:
Simply two chunks to me merged?

I wrote this so we later can make weekly/monthly/yearly reports with the views for the webui to display.